core/src/main/java/com/scalar/db/exception/storage/ExecutionException.java

```diff
@@ -1,12 +1,58 @@
 package com.scalar.db.exception.storage;
 
+import com.scalar.db.api.AuthAdmin;
+import java.util.Optional;
+import javax.annotation.Nullable;
+
 public class ExecutionException extends Exception {
 
+  private final boolean authenticationError;
+  private final boolean authorizationError;
+  private final boolean superuserRequired;
+  @Nullable private final AuthAdmin.Privilege requiredPrivilege;
```

Comment:
Added flags to the exceptions to indicate authentication and authorization errors. Initially, I considered introducing new exceptions for these errors, but to maintain backward compatibility, I opted for this approach.